File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        StringBuilder answer = new StringBuilder();
        
        String arr[] = new String[numbers.length];
        
        for(int i = 0; i < arr.length; i++){
            arr[i] = "" + numbers[i];
        }
        
        Arrays.sort(arr,(o1,o2) -> (o2 + o1).compareTo(o1 + o2));
        //  문자열 더한거로 비교
        
        
        if(arr[0].equals("0")){
            return "0";
        }
        
        for(int i = 0; i < arr.length; i++){
            answer.append(arr[i]);
        }
        return answer.toString();
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    answer = ''
    sorted_numbers = sorted(map(str, numbers), key = lambda x : x * 3, reverse = True)
    # print(sorted_numbers)
    answer = ''.join(sorted_numbers)
    if sorted_numbers[0] == '0':
        return '0'
    return answer